packages/components/src/components/post-megadropdown-trigger/post-megadropdown-trigger.tsx

```diff
@@ -1,29 +1,78 @@
+import { Component, Element, Prop, h, Host, State, Watch } from '@stencil/core';
+import { version } from '@root/package.json';
+import { checkType, getRoot } from '@/utils';
 
 @Component({
   tag: 'post-megadropdown-trigger',
-  shadow: true,
-  styleUrl: './post-megadropdown-trigger.scss',
+  styleUrl: 'post-megadropdown-trigger.scss',
+  shadow: false,
 })
 export class PostMegadropdownTrigger {
   /**
-   * Emits after each toggle
+   * ID of the mega dropdown element that this trigger is linked to. Used to open and close the specified mega dropdown.
    */
-  @Event() postToggle: EventEmitter;
+  @Prop() for!: string;
 
-  @State() toggled = false;
+  @Element() host: HTMLPostMegadropdownTriggerElement;
 
-  private handleClick() {
-    this.toggled = !this.toggled;
-    this.postToggle.emit();
+  /**
+   * Manages the accessibility attribute `aria-expanded` to indicate whether the associated mega dropdown is expanded or collapsed.
+   */
+  @State() ariaExpanded: boolean = false;
+
+  /**
+   * Reference to the slotted button within the trigger, if present.
+   * Used to manage click and key events for mega dropdown control.
+   */
+  private slottedButton: HTMLButtonElement | null = null;
+  private root?: Document | ShadowRoot;
+
+  /**
+   * Watch for changes to the `for` property to validate its type and ensure it is a string.
+   * @param forValue - The new value of the `for` property.
+   */
+  @Watch('for')
+  validateControlFor(forValue = this.for) {
+    checkType(forValue, 'string', 'The "for" property is required and should be a string.');
+  }
+
+  private get megadropdown(): HTMLPostMegadropdownElement | null {
+    const ref = this.root.getElementById(this.for);
+    return ref && ref.localName === 'post-megadropdown'
+      ? (ref as HTMLPostMegadropdownElement)
+      : null;
+  }
+
+  private handleToggle() {
+    const megadropdown = this.megadropdown;
+    if (megadropdown && this.slottedButton) {
+      this.ariaExpanded = !this.ariaExpanded;
+      this.slottedButton.setAttribute('aria-expanded', this.ariaExpanded.toString());
+      megadropdown.toggle(this.host);
+    } else {
+      console.warn(`No post-megadropdown found with ID: ${this.for}`);
+    }
+  }
+
+  componentDidLoad() {
+    this.root = getRoot(this.host);
+    this.validateControlFor();
+
+    this.slottedButton = this.host.querySelector('button');
+    if (this.slottedButton) {
+      this.slottedButton.setAttribute('aria-haspopup', 'menu');
+      this.slottedButton.addEventListener('click', () => {
+        this.handleToggle();
+      });
+    } else {
+      console.warn('No button found within post-megadropdown-trigger');
+    }
   }
 
   render() {
     return (
-      <Host slot="post-megadropdown-trigger">
-        <button aria-expanded={this.toggled} onClick={() => this.handleClick()}>
-          <slot></slot>
-        </button>
+      <Host data-version={version} tab-index="-1">
+        <slot></slot>
```

Comment:
Since we need a `button` element here, wouldn't it be possible to implement the button directly into the component and use a prop to let the users define the label of it?
This would not be an accessibility problem imo, because it's set up as a non-shadow component.

Reply:
I've move it there but I kept the slot for the label instead of adding the prop. What do you think?

Reply:
I can't see this change? I mean, there is no button wrapper around the slot.

Reply:
If you click on the file from the CR, you don't see it but if you go to the list of files, you should be able to see it.